src/libtls/tls_hkdf.h

```diff
@@ -136,6 +136,15 @@ struct tls_hkdf_t {
 	 */
 	bool (*resume)(tls_hkdf_t *this, chunk_t messages, chunk_t nonce, chunk_t *psk);
 
+	/**
+	 * Generate a PSK binder.
+	 *
+	 * @param seed				transcript-hash of client_hello to seed the PRF
```

Comment:
Might be worth adding a note (e.g. via separate `@note` tag) regarding the transcript hash (with reference to section 4.2.11.2 of the RFC), as the ClientHello is truncated and hashed without the actual binders (but with length headers adjusted for binders of the correct length).

Do we actually know the length of the PSK (i.e. the output length of the PRF) before calling this (or rather when generating ClientHello)?

Reply:
I added a note to reference to RFC 8446, section 4.2.11.2.

> Do we actually know the length of the PSK (i.e. the output length of the PRF) before calling this (or rather when generating ClientHello)?

Indirectly yes: `HKDF` is always initialised with a `hash_algorithm` and RFC 8446 only defines _SHA256_ (32 bytes) and _SHA384_ (48 bytes) as valid hash algorithms for the HKDF. For each hash a one byte length specifier is added and for all hashes a two byte length specifier. I added this information also in the `@note` documentation. Do you agree with that?

Reply:
I think I'd prefer adding a getter for the PSK length (i.e. `prf_t::get_block_size()`) to `hkdf_t` later, instead of deciding that outside of it. Alternatively, `binder()` could be changed to return the length and write directly to a buffer instead of allocating a chunk (we could then write directly to the actual location of the binder in the final message). Then it could just return the length when called with a NULL destination pointer (some crypto libraries use that approach). But let's leave it as it is and reconsider when we see how it fits with the actual resumption stuff.